01-ruby-fundamentals/07-oo-classes.md

```diff
@@ -63,50 +111,112 @@ Chair.new # => #<Chair:0x007fb7da550ca0> - this is an instance of a chair, it ha
-Chair.new.class # => Chair
+first_chair = Chair.new
+second_chair = Chair.new
+
+first_chair.name     # => Terrible Chair
+second_chair.name    # => Terrible Chair
 ```
-__Question: What are the limitations of our `Chair` _class_?__
 
-### `initialize`
-Let's start over building our class, but let's build it to be a little customizable. We want to be able to customize the attributes `color` & `name` when we create our chair.
+### Adding State Through Instance Variables
+
+Ruby allows us to attach variables to an instance of a class. Such variables are appropriately named _instance variables_. To create or use an instance variable, prefix the variable's name with the `@` symbol. Instance variables can be read and written in any instance method.
+
+By default, instance variables are not accessible outside the instance they're attached to. By convention, to make them readable we define a method with the same name as the variable, which returns the current value of that variable. This method is called a _reader_, _getter_ or _accessor_, depending on what language you're in (Ruby uses _reader_).
+
+Similarly, Ruby allows us to define a method that will allow the variable to be set with the `=` sign. This method is variously called a _writer_, _setter_ or _mutator_ (Ruby uses _writer_).
+
+Let's rewrite our `Chair` to take advantage of instance variables.
 
 ```ruby
 class Chair
-  def initialize
+  # reader for style
+  def style
+    return @style
+  end
 
+  # writer for style
+  def style=(new_style)
+    @style = new_style
+  end
+
+  def color
+    return @color
+  end
+
+  def color=(new_color)
+    @color = new_color
   end
 end
 ```
 
-That `initialize` method (often called a _constructor_) is called automatically from within the `new` method. So we will use it to "construct" our object. Since `initialize` is a method we create, we can define _parameters_ for it and pass _arguments_ to the `new` method. `new` is kind enough to pass along its _arguments_ to `initialize`.
+Now we can have different chairs with different attributes.
+
+```ruby
+kari_chair = Chair.new
+charles_chair = Chair.new
+
+kari_chair.style = "swivel"
+kari_chair.color = "black"
+
+charles_chair.style = "comfy"
+charles_chair.color = "green"
+
+puts kari_chair.style     # => swivel
+puts charles_chair.style  # => comfy
+```
 
-With those _parameters_, we can create _instance variables_. By prefixing a variable name with an `@` symbol, we create a variable that has visibility within __any instance method of the class__. As long as that specific instance of the class is around, Ruby will retain the value of its _instance variables_. Let's alter our `Chair` _class_ to look like this:
+Methods that access instance variables don't have to be simple getters or setters - they can have complex logic like any other method. For example, we might have `@seat_height` and `@max_height` instance variables and an `adjust_seat` method:
+
+```ruby
+def adjust_seat(direction, amount)
```

Comment:
I really like the idea of ensuring that we can show that there are methods other than the getters/setters that we may want to have within the class. I feel like this example is way too complex and is using variables that haven't been defined or set anywhere else (`@seat_height`). How about a simple "print" function that returns a nicely formatted string with the values from the instance variables above?